go-ethereum-guides/write-basic-quiz-dapp-in-go/write-basic-quiz-dapp-in-go.md

```diff
@@ -0,0 +1,1069 @@
+We'll start writing our Go DApp by initializing a connection to
+the Rinkeby network, using the Infura.io gateway endpoint that we 
+[set up earlier](#set-up-rinkeby-testnet-endpoint-on-infuraio).
+
+In your project directory,
+create a new `main.go` file and add the following code:
+
+```go
+package main
+
+import (
+    "context"
+    "log"
+    "fmt"
+
+    "github.com/ethereum/go-ethereum"
+    "github.com/joho/godotenv"
+)
+
+var myenv map[string]string
+
+const envLoc = ".env"
+
+func loadEnv() {
+	var err error
+	if myenv, err = godotenv.Read(envLoc); err != nil {
+		log.Printf("could not load env from %s: %v", envLoc, err)
+	}
+}
+
+func main(){
+    loadEnv()
+
+    ctx := context.Background()
+
+    client, err := ethclient.Dial(os.Getenv("GATEWAY"))
+    if err != nil {
+        log.Fatalf("could not connect to Ethereum gateway: %v\n", err)
+    }
+    defer client.Close()
+
+    accountAddress := common.HexToAddress("<enter_ethereum_address>")
+    balance, _ := client.BalanceAt(ctx, accountAddress, nil)
+    fmt.Printf("Balance: %d\n",balance)
+}
+```
+
+Replace `<enter_ethereum_address` with the address of the Ethereum account you've set up
+in [Set up an Ethereum account](#set-up-an-ethereum-account).
+
+Here, we're:
+
+- Loading data from our `.env` file into a map `myenv`
+using the `godotenv` package, which we've set as a
+requirement in our `go.mod` file.
+Our `.env` file should look like this now:
+
+    ```env
+    GATEWAY="https://rinkeby.infura.io/v3/<project_id>"
+    KEYSTORE="keystore/<keystore_filename>"
+    KEYSTOREPASS=""
+    ```
+    
+    - We can then access values set in our `.env` file with `myenv["KEYNAME"]`.
+    For example, access the `GATEWAY` value with `myenv["GATEWAY"]`.
+    - Notice that we've also written a function `loadEnv()` that we
+    can invoke at the beginning of every function scope. By placing
+    a `loadEnv()` call at the start of every function that
+    uses environment variables, we make sure
+    that we catch any updates to our `.env` file while our application
+    is running.
+
+- Setting up a connection to our Infura.io Rinkeby gateway by
+calling `ethclient.Dial("<gateway_endpoint>")`.
+Note that this works for both TCP (HTTP/S) 
+and IPC (`<datadir>/geth.ipc`) endpoints.
+- Getting the balance of our Ethereum account by calling
+`client.GetBalance(ctx, accountAddress, nil)`.
+We have to convert our Ethereum address from a hex string like
+`48fddc985ecc605127f1a1c098c817778187637c` to the `common.Address`
+type before passing it to `GetBalance()`.
+- Printing the result of `GetBalance()`.
+
+Test your application by running `go run main.go` in the terminal.
+If it prints the balance of your Ethereum account,
+your application has successfully loaded configuration from your `.env` file
+and sent a message call to the Rinkeby network 🎉.
+
+Now that we know that our `ethclient.Dial()` call works,
+we won't need the `GetBalance()` call. Remove it from `main()`.
+
+### Create session
```

Comment:
@zeddee Is the reader meant to add these code snippets anywhere, or is this just for example?

Reply:
@zeddee pingping ^^

Reply:
sorry! didn't see this one. how about:

----
[...] `Getbalance()` call. Remove it from your Go file, such that your `main()` block looks like this:

```
func main(){
    loadEnv()

    ctx := context.Background()

    client, err := ethclient.Dial(os.Getenv("GATEWAY"))
    if err != nil {
        log.Fatalf("could not connect to Ethereum gateway: %v\n", err)
    }
    defer client.Close()
}
```
----